src/nfa/nfa-complement.cc

```diff
@@ -25,46 +25,35 @@ using namespace Mata::Util;
 Nfa Mata::Nfa::Algorithms::complement_classical(
 	const Nfa&         aut,
 	const Alphabet&    alphabet,
-	std::unordered_map<StateSet, State>* subset_map)
+	bool minimize_during_determinization)
 { // {{{
  	Nfa result;
-
- 	bool delete_subset_map = false;
-	if  (nullptr == subset_map)
-	{
-		subset_map = new std::unordered_map<StateSet, State>();
-		delete_subset_map = true;
-	}
-
-	result = determinize(aut, subset_map);
-	State sink_state = result.add_state();
-	auto it_inserted_pair = subset_map->insert({{}, sink_state});
-	if (!it_inserted_pair.second)
-	{
-		sink_state = it_inserted_pair.first->second;
+	State sink_state;
+	if (minimize_during_determinization) {
+		result = minimize_brzozowski(aut); // brzozowski minimization makes it deterministic
+		if (result.final.size() == 0) {
+			// if automaton does not accept anything, then there is only one (initial) state
+			// which can be the sink state (so we do not create unnecessary one)
+			sink_state = *result.initial.begin();
```

Comment:
Can you be sure that there will be at least one initial state? I do not think so. This would then produce a segmentation fault.

Reply:
I consider this resolved then.